File: java_sdk/src/main/java/business_api_client/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2024-11-01T15:21:34.532937-07:00[America/Los_Angeles]")
public class OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet {
  @JsonProperty("operator")
  private String operator = null;

  @JsonProperty("rules")
  private List<OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> rules = new ArrayList<OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules>();

  public OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet operator(String operator) {
    this.operator = operator;
    return this;
  }

   /**
   * Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.
   * @return operator
  **/
  @Schema(required = true, description = "Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.")
  public String getOperator() {
    return operator;
  }

  public void setOperator(String operator) {
    this.operator = operator;
  }

  public OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet rules(List<OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> rules) {
    this.rules = rules;
    return this;
  }

  public OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet addRulesItem(OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules rulesItem) {
    this.rules.add(rulesItem);
    return this;
  }

   /**
   * Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.
   * @return rules
  **/
  @Schema(required = true, description = "Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.")
  public List<OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> getRules() {
    return rules;
  }

  public void setRules(List<OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules> rules) {
    this.rules = rules;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet openApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet = (OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet) o;
    return Objects.equals(this.operator, openApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.operator) &&
        Objects.equals(this.rules, openApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.rules);
  }

  @Override
  public int hashCode() {
    return Objects.hash(operator, rules);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet {\n");
    
    sb.append("    operator: ").append(toIndentedString(operator)).append("\n");
    sb.append("    rules: ").append(toIndentedString(rules)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';
import {OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules} from './OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules.js';

/**
 * The OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet model module.
 * @module model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet
 * @version 0.1.2
 */
export class OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet {
  /**
   * Constructs a new <code>OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet</code>.
   * @alias module:model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet
   * @class
   * @param operator {String} Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.
   * @param rules {Array.<module:model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules>} Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.
   */
  constructor(operator, rules) {
    this.operator = operator;
    this.rules = rules;
  }

  /**
   * Constructs a <code>OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet} obj Optional instance to populate.
   * @return {module:model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet} The populated <code>OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet();
      if (data.hasOwnProperty('operator'))
        obj.operator = ApiClient.convertToType(data['operator'], 'String');
      if (data.hasOwnProperty('rules'))
        obj.rules = ApiClient.convertToType(data['rules'], [OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules]);
    }
    return obj;
  }
}

/**
 * Required when exclusion_rule_set is passed. Operator between the exclusion rules within the exclusion rule set. Enum value: OR.
 * @member {String} operator
 */
OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.prototype.operator = undefined;

/**
 * Required when exclusion_rule_set is passed. A list of exclusion rules within the exclusion rule set.
 * @member {Array.<module:model/OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSetRules>} rules
 */
OpenApiv13dmpcustomAudiencerulecreateRuleSpecExclusionRuleSet.prototype.rules = undefined;

